tests/integration/suite/daprd/lock/http/errors.go

```diff
@@ -0,0 +1,520 @@
+/*
+Copyright 2023 The Dapr Authors
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+    http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implieh.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package http
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/http"
+	"strings"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+
+	"github.com/dapr/dapr/tests/integration/framework"
+	"github.com/dapr/dapr/tests/integration/framework/process/daprd"
+	"github.com/dapr/dapr/tests/integration/framework/util"
+	"github.com/dapr/dapr/tests/integration/suite"
+	kitErrors "github.com/dapr/kit/errors"
+)
+
+const (
+	ErrInfoType      = "type.googleapis.com/google.rpc.ErrorInfo"
+	ResourceInfoType = "type.googleapis.com/google.rpc.ResourceInfo"
+	BadRequestType   = "type.googleapis.com/google.rpc.BadRequest"
+	HelpType         = "type.googleapis.com/google.rpc.Help"
+	LockStoreName    = "lockstore"
+)
+
+func init() {
+	suite.Register(new(errorcodes))
+}
+
+type errorcodes struct {
+	daprd *daprd.Daprd
+}
+
+func (e *errorcodes) Setup(t *testing.T) []framework.Option {
+	e.daprd = daprd.New(t,
+		daprd.WithResourceFiles(`
+apiVersion: dapr.io/v1alpha1
+kind: Component
+metadata:
+  name: lockstore
+spec:
+  type: lock.redis
+  version: v1
+  metadata:
+  - name: redisHost
+    value: localhost:6379
+  - name: redisPassword
+    value: ""
+`))
+
+	return []framework.Option{
+		framework.WithProcesses(e.daprd),
+	}
+}
+
+func (e *errorcodes) Run(t *testing.T, ctx context.Context) {
+	e.daprd.WaitUntilRunning(t, ctx)
+
+	httpClient := util.HTTPClient(t)
+
+	// Covers apierrors.ERR_LOCK_NOT_FOUND
+	t.Run("lock doesn't exist", func(t *testing.T) {
+		name := "lock-doesn't-exist"
+		payload := `{"resourceId": "resource", "lockOwner": "owner", "expiryInSeconds": 10}`
+		endpoint := fmt.Sprintf("http://localhost:%d/v1.0-alpha1/lock/%s", e.daprd.HTTPPort(), name)
+
+		req, err := http.NewRequestWithContext(ctx, http.MethodPost, endpoint, strings.NewReader(payload))
+		require.NoError(t, err)
+
+		resp, err := httpClient.Do(req)
+		require.NoError(t, err)
+
+		require.Equal(t, "application/json", resp.Header.Get("Content-Type"))
+		require.Equal(t, http.StatusNotFound, resp.StatusCode)
```

Comment:
In looking at the `predefined.go` I believe this should be the legacy http code of: `http.StatusBadRequest`. By changing this it might break existing users, we want to keep the logic as is, even if it's not correct. You can add a `// TODO: change this to be http.StatusNotFound in 2 releases since it would be a breaking change` to the `errors/lock.go`